runners/spark/src/test/java/org/apache/beam/runners/spark/aggregators/metrics/sink/InMemoryMetrics.java

```diff
@@ -45,9 +49,15 @@ public InMemoryMetrics(final Properties properties,
   public static <T> T valueOf(final String name) {
     final T retVal;
 
+    // this might fail in case we have multiple aggregators with the same suffix after
```

Comment:
This will happen if aggregators are assigned with the same name, right ?
Doesn't bother me for testing, we should think (generally) how/if we're able to avoid/alert/handle this case, but this is out-of-scope for this PR.